### PR TITLE
Massive Refactor for jest testing

### PR DESCRIPTION
**Pull request by O-Mutt**
This modifies all the tests to run under jest. There are a lot of broken tests still.

Ref:
#1136 #1137

### CODE REVIEW

File: libs/feedback/jest.conf.js [hunk 1]
```diff
@@ -1,6 +1,10 @@
-// Karma configuration file, see link for more information
-// https://karma-runner.github.io/1.0/config/configuration-file.html
+module.exports = {
+  name: 'feedback',
+  preset: '../../jest.config.js',
+  coverageDirectory: '../../coverage/libs/feedback'
+};
 
+/*
```
Comment (kirjs) [inline libs/feedback/jest.conf.js]: Drop the config?

File: apps/kirjs/src/app/modules/binary/binary-plain/binary-plain.component.spec.ts [hunk 2]
```diff
@@ -1,20 +1,25 @@
 import { async, ComponentFixture, TestBed } from '@angular/core/testing';
 
 import { BinaryPlainComponent } from './binary-plain.component';
+import {
+  CUSTOM_ELEMENTS_SCHEMA,
```
Comment (kirjs) [inline apps/kirjs/src/app/modules/binary/binary-plain/binary-plain.component.spec.ts]: Unused?

**Comment by kirjs** [comment]
wow, this looks awesome! 

Curious if you were in touch with @hansal7014 (from #1136) to avoid duplicate work?

File: libs/utils/jest.conf.js [hunk 3]
```diff
@@ -1,4 +1,10 @@
-module.exports = function(config) {
+module.exports = {
+  name: 'utils',
+  preset: '../../jest.config.js',
+  coverageDirectory: '../../coverage/libs/utils'
+};
+
+/*module.exports = function(config) {
```
[the remark this answers is not in the document]
Comment (Mutmatt) [reply libs/utils/jest.conf.js]: I was leaving those for when I was testing to make sure if there were imports/anything that i could use in the jest.config. I will clear them for sure

**Comment by Mutmatt** [comment]
> wow, this looks awesome!
> 
> Curious if you were in touch with @hansal7014 (from #1136) to avoid duplicate work?

Did not, i must have started before those comments came in. It took me a few days of messing around to get everything working well enough

**Comment by kirjs** [comment]
Ok, cool, good to know, I pinged the other thread to avoid doing the same stuff twice in the future, for now let's merge this in

**Comment by Mutmatt** [comment]
> Ok, cool, good to know, I pinged the other thread to avoid doing the same stuff twice in the future, for now let's merge this in

I'm pulled the travis changes because there are still a ton of broken tests.

**Comment by NothingEverHappens** [comment]
Just wanted to send a friendly ping to understand the state of this and make sure it's not blocked from our side.

**Comment by Mutmatt** [comment]
Hey @NothingEverHappens this was not blocked just haven't had a time to get back to the enormity of this. This in and of itself should be ok if we wanted to take a small chunk out of the project

**Comment by NothingEverHappens** [comment]
Cool, would you have some time to address the feedback? (I think it's mostly dropping unused code bytes, and other small things)?

I think this is a big step forward compered to what we have now and would be awesome to get merged.

**Comment by NothingEverHappens** [comment]
Hey @Mutmatt, are you planning to look into merging this in?
Or alternatively I can work on doing this.

**Pull request closed without merging**